src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -525,6 +582,388 @@ public List<UserType> fetchUserTypes() {
+        }
+        return status;
+    }
+
+    private Optional<Object> getUserProfileUpdateResponse(UserProfileUpdatedData userProfileUpdatedData,
+                                                          String userId, String origin) {
+        Response response = userProfileFeignClient.modifyUserRoles(userProfileUpdatedData, userId, origin);
+        log.info("{}:: UserProfile update roles :: status code {}", loggingComponentName,
+                response.status());
+
+        ResponseEntity<Object> responseEntity = toResponseEntity(response, UserProfileRolesResponse.class);
+
+        return validateAndGetResponseEntity(responseEntity);
+    }
+
+    private Optional<Object> validateAndGetResponseEntity(ResponseEntity<Object> responseEntity) {
+        if (nonNull(responseEntity)) {
+            return Optional.ofNullable(responseEntity.getBody());
+        }
+        return Optional.empty();
+    }
+
+    // update roles in sidam and filter if failed in User profile
+    public List<CaseWorkerProfile> updateSidamRoles(List<CaseWorkerProfile> updateCaseWorkerProfiles,
+                                                    StaffProfileCreationRequest cwUiRequest) {
+        List<CaseWorkerProfile> filteredUpdateCwProfiles = new ArrayList<>();
+        for (CaseWorkerProfile dbProfile : updateCaseWorkerProfiles) {
+            boolean isAddRoleSuccess = updateUserRolesInIdam(cwUiRequest,
+                    dbProfile.getCaseWorkerId());
+            if (isAddRoleSuccess) {
+                filteredUpdateCwProfiles.add(dbProfile);
+            }
+        }
+        return filteredUpdateCwProfiles;
+    }
+
+    public boolean updateUserRolesInIdam(StaffProfileCreationRequest cwrProfileRequest, String idamId) {
+
+        boolean result = false;
+        try {
+            Response response = userProfileFeignClient.getUserProfileWithRolesById(idamId);
+            ResponseEntity<Object> responseEntity = toResponseEntity(response, UserProfileResponse.class);
+
+            Optional<Object> resultResponse = validateAndGetResponseEntity(responseEntity);
+            if (resultResponse.isPresent() && resultResponse.get() instanceof UserProfileResponse profileResponse) {
+                if (isNotTrue(profileResponse.getIdamStatus().equals(STATUS_ACTIVE))) {
+
+                    staffProfileAuditService.saveStaffAudit(AuditStatus.FAILURE,IDAM_STATUS_NOT_ACTIVE,
+                            StringUtils.EMPTY,cwrProfileRequest,STAFF_PROFILE_UPDATE);
+
+                    log.info("{}:: updateUserRolesInIdam :: status code {}", loggingComponentName,
+                            profileResponse.getIdamStatus());
+
+                    throw new StaffReferenceException(HttpStatus.BAD_REQUEST, StringUtils.EMPTY,
+                            IDAM_STATUS_NOT_ACTIVE);
+                }
+
+            } else {
+                log.info("{}:: updateUserRolesInIdam :: status code {}", loggingComponentName,
+                        UP_FAILURE_ROLES);
```

Comment:
Please log it as an error and please audit the failure

Reply:
done